source/common/network/io_socket_handle_impl.cc

```diff
@@ -69,6 +110,53 @@ Api::IoCallUint64Result IoSocketHandleImpl::writev(const Buffer::RawSlice* slice
   return sysCallResultToIoCallResult(result);
 }
 
+Api::IoCallUint64Result IoSocketHandleImpl::sendto(const Buffer::RawSlice& slice, int flags,
+                                                   const Address::Instance& address) {
+  sockaddr_storage ss;
+  socklen_t ss_len = sizeof ss;
+  getSocketAddressInfo(address, ss, ss_len);
+
+  auto& os_syscalls = Api::OsSysCallsSingleton::get();
+  const Api::SysCallSizeResult result = os_syscalls.sendto(
+      fd_, slice.mem_, slice.len_, flags, reinterpret_cast<sockaddr*>(&ss), ss_len);
```

Comment:
nit: use Address::InstanceBase::sockAddr() and Address::InstanceBase::sockAddrLen() here directly. You may need to downcast `address`, but doing so you can get rid of getSocketAddressInfo()

Reply:
Not sure if we can do this in the current code because of  what we discussed previously in a different comment.

address_impl.cc includes "io_socket_handle_impl.h" and depends on the lib "io_socket_handle_lib".

If we try to include address_impl.h in io_socket_handle_impl.cc and add the dependency, there is a circular dependency which breaks the build.

Reply:
I thought it would be just as easy as combining //source/common/netowrk:address_lib and //source/common/netowrk:io_socket_handle_lib. No? Maybe I missed some important facts.

Reply:
looking at the way things have been coded, I presumed that the coding style is to produce 1 lib per .cc file.
Combining the 2 cc files into a single lib is a way to achieve what we want - but is that okay? If so, I can do that

Reply:
I pushed an update with this change - let me know if it looks good.

Reply:
Looks good. Thanks for making the change!